src/main/groovy/com/bmuschko/gradle/docker/tasks/AbstractDockerRemoteApiTask.groovy

```diff
@@ -51,23 +52,9 @@ abstract class AbstractDockerRemoteApiTask extends DefaultTask {
     @Internal
     ThreadContextClassLoader threadContextClassLoader
 
-    /**
-     * Closure to handle the possibly throw exception
-     */
-    @Internal
-    Closure onError
-
-    /**
-     * Closure to handle results
-     */
-    @Internal
-    Closure onNext
-
-    /**
-     * Closure to handle task completion
-     */
-    @Internal
-    Closure onComplete
+    private Action<? super Throwable> errorHandler
+    protected Action<? super Object> nextHandler
+    private Runnable completeHandler
```

Comment:
Why change the names?

Reply:
Because Groovy auto-generates getters/setter under the cover which we don't want. We want end users to only provide a handler through the exposed method and not some `setOnNext(...)`. There's also no point in having a getter for end users. Overall a cleaner API surface.

Reply:
Also ... why a Runnable over Action for the `onComplete`?

Reply:
The `complete` callback doesn't receive a parameter. At least at the moment. The only sensible generic type for an `Action` would be a `Void`. I also checked with the Gradle Kotlin DSL team and that's what they recommended.

Reply:
Gotcha. Because if you were to do a `Action<? super Void>` you _may_ be required to actually pass in a type/null to make it not complain?

Reply:
I don't think Groovy would complain really because of it being so dynamic. I guess if you would use a parameter the value would be `null`. Not sure how the Kotlin DSL would behave. Fact is...there's simply nothing we pass in so there's no reason to define a type of it. For more info see https://gradle-community.slack.com/archives/CAD95CR62/p1539355045000100.

Reply:
Also ... why the mix between private and protected here?

Reply:
Because the field `nextHandler` is used in subclasses whereas the other two fields are not.

Reply:
Oh, forgot to mention. Introducing these methods also doesn't require us to add `@Internal` anymore as there's no more getter method.